setup.py

```diff
@@ -68,10 +68,41 @@
         long_description = f.read()
 
 
+dependencies = [
+    "numpy>=1.24,<2.6",  # see #4773
+]
+optional_dependencies = {
+    "all": [
+        "scipy>=1.10,<1.17",  # see #4773
+        "Cython>=3",
+        "optuna>=2.0",
+    ],
+    "test": [
+        "packaging",
+        "pytest>=7.2",
+        "hypothesis>=6.37.2,<6.55.0",
+        "mpmath",
+    ],
+}
+if not ctx.use_hip:
+    dependencies.append("cuda-pathfinder>=1.3.2,<2.0a0")
```

Comment:
This is unsafe as using pre-release specifier will allow pip to pick pre-releases even without `--pre`.

```suggestion
    dependencies.append("cuda-pathfinder>=1.3.2,==1.*")
```

https://pip.pypa.io/en/stable/news/#id104:
> `<2.0dev` can include `1.0rc1`